### PR TITLE
seserializable_put

### PR DESCRIPTION
**Pull request by pham-van-tuy**
serializable_put fix put data class

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -2,6 +2,7 @@ plugins {
     id 'com.android.application'
     id 'kotlin-android'
     id 'kotlin-android-extensions'
+   // id 'kotlinx-serialization'
```
Comment (thuypt) [inline app/build.gradle:5]: seem this no longer use, let delete it

File: app/src/main/java/com/example/learn09082021v001/DataInput.kt [hunk 2]
```diff
@@ -9,32 +9,26 @@ import android.util.Log
 import androidx.activity.result.ActivityResult
 import androidx.activity.result.contract.ActivityResultContracts
 import kotlinx.android.synthetic.main.activity_data_input.*
+import kotlinx.android.synthetic.main.activity_main.*
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/DataInput.kt:12]: why import activity_main here?

File: app/src/main/java/com/example/learn09082021v001/MainActivity.kt [hunk 3]
```diff
@@ -9,64 +9,69 @@ import androidx.activity.result.ActivityResult
...
 import kotlinx.android.synthetic.main.activity_data_input.*
 import kotlinx.android.synthetic.main.activity_main.*
-public var nameAndroid = "nameAndroidView"
-public var scoreAndroid = "scoreAndroidView"
-public var nameIos = "nameIosView"
-public var scoreIos = "scoreIosView"
+import java.io.Serializable;
+
+public val sendData = "datatranfer"
+public var scoreOOP = NameScoreOOP()
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/MainActivity.kt:15]: this is variable, not a constant, should declare it inside MainActivity

File: app/src/main/java/com/example/learn09082021v001/MainActivity.kt [hunk 4]
```diff
@@ -9,64 +9,69 @@ import androidx.activity.result.ActivityResult
 import androidx.activity.result.contract.ActivityResultContracts
 import kotlinx.android.synthetic.main.activity_data_input.*
 import kotlinx.android.synthetic.main.activity_main.*
-public var nameAndroid = "nameAndroidView"
-public var scoreAndroid = "scoreAndroidView"
-public var nameIos = "nameIosView"
-public var scoreIos = "scoreIosView"
+import java.io.Serializable;
+
+public val sendData = "datatranfer"
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/MainActivity.kt:14]: use `const val DATA_TRANSFER = "data_transfer"
- typo error in datatranfer
- const use const val not public val
- const val should be capitalize

File: app/src/main/java/com/example/learn09082021v001/MainActivity.kt [hunk 5]
```diff
@@ -9,64 +9,69 @@ import androidx.activity.result.ActivityResult
...
+    fun clearData() {
         tvViewNameAndroid.setText("")
         tvViewNameIOS.setText("")
         tvViewScoreAndroid.setText("")
         tvViewScoreIOS.setText("")
-        btnStartReset.setText("START") }
+        btnStartReset.setText("START")
+    }
+
+    fun getdataclassOOP() {
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/MainActivity.kt:71]: this function do update data to view, its name can be `updateDataToUI` or `updateUI`

File: app/src/main/java/com/example/learn09082021v001/NameScoreOOP.kt [hunk 6]
```diff
@@ -25,11 +26,12 @@ class NameScoreOOP (
     }
 
 }
-class datasaveandload
-{
+
+class datasaveandload {
```
Comment (thuypt) [inline app/src/main/java/com/example/learn09082021v001/NameScoreOOP.kt:30]: don't naming a class like this. Check https://developer.android.com/kotlin/style-guide#naming

File: app/src/main/java/com/example/learn09082021v001/MainActivity.kt (hunk 3 above)
Comment (pham-van-tuy) [reply app/src/main/java/com/example/learn09082021v001/MainActivity.kt:15]: because inside Mainactivity only use Mainactivity : i need use 3 activity

**Review by thuypt — APPROVED** [review]

**Pull request merged**